perfact/zodbsync/tests/test_sync.py

```diff
@@ -979,3 +980,44 @@ def test_create_structure_and_reset_commits(self):
         self.run('reset', 'HEAD~1')
         assert folder_1 not in self.app.objectIds()
         assert not os.path.isfile(self.meta_file_path(folder_1))
+
+    @pytest.mark.parametrize('meta_type', object_types.object_handlers)
+    def test_objecttypes(self, meta_type):
+        """
+        Generic test that is executed for each coded object type. This creates
+        an object and writes a modification to it, without actually checking
+        for anything. Some are known to fail, for example because they need
+        products that are not published on pypi or because they need external
+        ressources like non-free libraries for external data connections.
+        """
+        if meta_type in ['DTML TeX', 'ZForce', 'External Method',
+                         'Z cxOracle Database Connection',
+                         'Z sap Database Connection']:
+            pytest.xfail()
+
+        if 'Test' not in self.app.objectIds():
+            self.app.manage_addProduct['OFSP'].manage_addFolder(id='Test')
+        if meta_type in ['User Folder', 'Simple User Folder']:
+            objid = 'acl_users'
+        else:
+            objid = 'testobj'
+        parent = self.app.Test
+        handler = object_types.object_handlers[meta_type]
+        # data that is required by some objects and ignored by others
+        add_data = {
+            'title': 'test',
+            'content_type': 'text/plain',
+            'connection_id': 'dbconn',
+            'connection_string': '',
+            'autocommit': False,
+            'maxrows': 100,
+            'args': '',
+            'source': '',
+            'smtp_host': 'localhost',
+            'smtp_port': '25',
+        }
```

Comment:
Required by some, ignored by most is okay, I guess, but kind of hacky.

I'd still prefer a dictionary of dictionaries, naming explicitly which additional arguments are required for which meta_type.

Reply:
I think I would rather not have a large dict of dicts in the test. Maybe the handlers in `object_types.py` could define a "default dictionary" or even simply use default values in their `write` methods, not expecting anything except for the title in `data`.
But I would like to postpone this to a future refactoring, in particular because #85 builds on this (although it could be rebased, there is no logical dependency).

Reply:
Agreed.